include/librealsense2/dds/dds-device.cpp

```diff
@@ -36,7 +74,180 @@ class dds_device::impl
+        {
+            if( _reader->wait_for_unread_message( { 1, 0 } ) )
+            {
+                dds::topics::raw::device::notification raw_data;
+                topics::device::notification data;
+                SampleInfo info;
+                // Process all the samples until no one is returned,
+                // We will distinguish info change vs new data by validating using `valid_data`
+                // field
+                while( ReturnCode_t::RETCODE_OK == _reader->take_next_sample( &raw_data, &info ) )
+                {
+                    // Only samples for which valid_data is true should be accessed
+                    // valid_data indicates that the instance is still ALIVE and the `take` return
+                    // an updated sample
+                    if( info.valid_data )
+                    {
+                        data = raw_data;
+                        auto msg_id = static_cast< topics::device::notification::msg_type >( raw_data.id() );
+                        switch( msg_id )
+                        {
+                        case topics::device::notification::msg_type::DEVICE_HEADER:
+                            if( state_type::WAIT_FOR_DEVICE_HEADER == state )
+                            {
+                                num_of_sensors = data.get< topics::device::notification::device_header_msg >()->num_of_sensors;
+                                LOG_INFO( "got DEVICE_HEADER message with " << num_of_sensors << " sensors" );
+                                state = state_type::WAIT_FOR_SENSOR_HEADER;
+                            }
+                            else
+                                LOG_ERROR( "Wrong message received, got 'DEVICE_HEADER' when the state is: " << state );
+                            break;
+                        case topics::device::notification::msg_type::SENSOR_HEADER: {
+                            if( state_type::WAIT_FOR_SENSOR_HEADER == state )
+                            {
+                                auto sensor_header = data.get< topics::device::notification::sensor_header_msg >();
+                                LOG_INFO( "got SENSOR_HEADER message for sensor: " << sensor_header->name << " of type: "
+                                          << ( sensor_header->type == topics::device::notification::sensor_header_msg::sensor_type::VIDEO_SENSOR
+                                                   ? "VIDEO_SENSOR"
+                                                   : "MOTION_SENSOR" ) );
+
+                                sensor_index_to_name.emplace( sensor_header->index, sensor_header->name );
+                                state = state_type::WAIT_FOR_PROFILES;
+                            }
+                            else
+                                LOG_ERROR( "Wrong message received, got 'SENSOR_HEADER' when the state is: " << state );
+                        }
+                        break;
```

Comment:
I suggest this break be put inside the {}

Reply:
I removed the {} as they are redundant here

Reply:
Done